internal/local/structure_v1.go

```diff
@@ -110,3 +113,115 @@ func (a *AppDataV1) LoadData(rootDir string) error {
 	a.Services = services
 	return nil
 }
+
+// ConfigData marshals the config data out to JSON
+func (a AppDataV1) ConfigData() ([]byte, error) {
+	temp := &struct {
+		ConfigVersion        realm.AppConfigVersion `json:"config_version"`
+		ID                   string                 `json:"app_id,omitempty"`
+		Name                 string                 `json:"name"`
+		Location             realm.Location         `json:"location"`
+		DeploymentModel      realm.DeploymentModel  `json:"deployment_model"`
+		Environment          string                 `json:"environment,omitempty"`
+		Security             map[string]interface{} `json:"security"`
+		Hosting              map[string]interface{} `json:"hosting,omitempty"`
+		CustomUserDataConfig map[string]interface{} `json:"custom_user_data_config"`
+		Sync                 map[string]interface{} `json:"sync"`
+	}{
+		ConfigVersion:        a.ConfigVersion(),
+		ID:                   a.ID(),
+		Name:                 a.Name(),
+		Location:             a.Location(),
+		DeploymentModel:      a.DeploymentModel(),
+		Environment:          a.Environment,
+		Security:             a.Security,
+		CustomUserDataConfig: a.CustomUserDataConfig,
+		Sync:                 a.Sync,
+	}
+	return MarshalJSON(temp)
+}
+
+// WriteData will write the local Realm app data to disk
+func (a *AppDataV1) WriteData(rootDir string) error {
+	if err := writeSecrets(rootDir, a.Secrets); err != nil {
+		return err
+	}
+	if err := writeEnvironments(rootDir, a.Environments); err != nil {
+		return err
+	}
+	if err := writeValues(rootDir, a.Values); err != nil {
+		return err
+	}
+	if err := writeGraphQL(rootDir, a.GraphQL); err != nil {
+		return err
+	}
+	if err := writeServices(rootDir, a.Services); err != nil {
+		return err
+	}
+	if err := writeFunctionsV1(rootDir, a.Functions); err != nil {
+		return err
+	}
+	if err := writeAuthProviders(rootDir, a.AuthProviders); err != nil {
+		return err
+	}
+	if err := writeTriggers(rootDir, a.Triggers); err != nil {
+		return err
+	}
+	return nil
+}
+
+func writeFunctionsV1(rootDir string, functions []map[string]interface{}) error {
+	if err := os.MkdirAll(filepath.Join(rootDir, NameFunctions), os.ModePerm); err != nil {
+		return err
+	}
+	for _, function := range functions {
+		configFunction := function[NameConfig].(map[string]interface{})
```

Comment:
if/when we _have_ to use these casts, use the `var, ok` syntax and return an error if `!ok` to avoid the potential of panics